### ISSUE TITLE
Issue Upgrading to .NetCore 3.1?

### ISSUE THREAD

**Issue by gsbanner13** [OrchardCMS/OrchardCore#5766]
Hello!  As part of social distancing, I have been working from home.  So, I tried to bring over the code for the OrchardCore site I am building for us.  I donwloaded .netcore 3.1 sdk and then opened my solution in VS 2019 (same as I use for work). Once I did, all of my OrchardCore references began throwing errors because they were not compatible with 3.1.  So, I used NuGet to upgrade them all.  Once this was done, the site would still not build.  It was telling me I was missing the Index Model.  I do not use an Index.cshtml (it is excluded), instead using the Layout.Liquid that is located in a Custom Theme project that I reference in the main project.  If I add the Index.cshtml back in, I now get a new error saying that the Layout.Liquid cannot be found.  Any ideas or suggestions?

**Comment by deanmarcussen** [comment]
Can you tell us which version of Orchard you were using, and which version you have tried to upgrade to?

**Comment by gsbanner13** [comment]
Hmmm...unsure.  How can I check?  all of my references say Version="1.0.0-rc1-12410" but I'm not sure if that is the same as the overall OrchardCore version?

**Comment by infofromca** [comment]
@gsbanner13 I had this #5146

**Comment by gsbanner13** [comment]
My machine at work is running 16.4.4.  When I have old packages and .net core 3.0 everything works.  If I switch to .net core 3.1 and update packages accordingly, site no longer works.

My PC at home is running VS 16.5.0

Are there compatibility issues with > 16.4.0?

**Comment by deanmarcussen** [comment]
Did you update your `.csproj` files appropriately to target the `netcoreapp3.1` framework ?

e.g.
```
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>netcoreapp3.1</TargetFramework>
```

This is necessary in your host project, themes, and modules.

VS 16.5.0 is fine

**Comment by Skrypt** [comment]
`<TargetFramework>$(AspNetCoreTargetFramework)</TargetFramework>`

Is the way to do this now. It will upgrade automatically.

**Comment by deanmarcussen** [comment]
> <TargetFramework>$(AspNetCoreTargetFramework)</TargetFramework>

That variable is only available for OC Source Code, it isn't exposed for use with nuget packages. (I think from one time when I tried to use it)

**Comment by gsbanner13** [comment]
Sorry for the lack of follow up, but I got pulled in to some other projects over the last 10 days or so.  I finally tracked down the problem.  In my main site references, I was still referencing OrchardCore.Appllication.CMS.Targets.  I had to removed this reference and instead reference OrchardCore.Appllication.CMS.**Core.**Targets.  When I did this AND made sure my Theme and Module projects were referencing .Net 3.1 AND updated all appropriate OrchardCore references, I was able to build.
